solver/src/driver/solver_settlements.rs

```diff
@@ -76,14 +76,23 @@ pub struct RatedSettlement {
     pub settlement: SettlementWithSolver,
     pub surplus: BigRational,
     pub gas_estimate: U256,
+    pub gas_price: BigRational,
 }
 
 impl RatedSettlement {
-    pub fn objective_value(&self, gas_price: f64) -> BigRational {
-        let gas_price = BigRational::from_float(gas_price).unwrap();
+    pub fn objective_value(&self) -> BigRational {
         let gas_estimate = self.gas_estimate.to_big_rational();
+        Self::compute_objective_value(&self.surplus, &gas_estimate, &self.gas_price)
+    }
+
+    // Helper function to allow unit testing objective value computation without a Settlement.
+    fn compute_objective_value(
```

Comment:
Does having this method separate still make sense? feels like it could be on RatedSettlement but no strong feelings if this makes testing it much easier. If you want to keep the method, it should probably become a free method.

Reply:
Now that I see the term RatedSettlement mentioned here, I might also suggested calling it `RankedSettlement`

Reply:
It allows testing "RatedSettlement.objective_value()" without having to build a "RatedSettlement" object, which requires passing in a settlement. This is used in the test I added. Note that this is a static method (doesn't have self) so essentially a free method namespaced on RatedSettlement. I'm happy to move it to outside the class if you like.

Reply:
You are right that the free vs static doesn't make a big difference (especially since it's private). I'd go with the general style of the repo which is using free methods over static methods when in doubt.

Reply:
Personally I prefer free methods for these kinds of helper functions. For me associated functions (i.e. static functions) are typically for constructing instances of a type.

This is just personal preference. I imagine the compiler treats them in the exact same way.